utils/fixed_decimal/src/decimal.rs

```diff
@@ -1916,12 +1912,8 @@ impl FromStr for FixedDecimal {
         if has_exponent {
             let mut pow = 0;
             let mut pos_neg = 1;
-            let no_sign_str_from_exponent = match no_sign_str.get(exponent_index + 1..) {
-                Some(slice) => slice,
-                None => unreachable!("`exponent_index + 1` must be within range."),
-            };
-
-            for digit in no_sign_str_from_exponent {
+            #[allow(clippy::indexing_slicing)] // exponent_index comes from enumerate
```

Comment:
Just because `exponent_index` is in bounds doesn't mean `exponent_index + 1` is as well

Reply:
now, the comment is more explanatory.